src/components/common/controllers/drag.ts

```diff
@@ -267,16 +313,30 @@ class DragController implements ReactiveController {
     this._updatePosition(event);
     this._updateMatcher(event);
 
-    const parameters = { event, state: this._stateParameters };
+    this._direction = this._trackPointerMovement(event.clientX, event.clientY);
+
+    const parameters = {
+      event,
+      state: this._stateParameters,
+      direction: this._direction,
+    };
+
     this._options.move?.call(this._host, parameters);
 
     this._assignPosition(this._dragItem);
   }
 
   private _handlePointerEnd(event: PointerEvent): void {
+    this._previousPointerPosition = null;
```

Comment:
Maybe just initialize the previous coordinates as `{x: 0, y: 0}` on initialization of the controller and then just updated through the pointerdown -> pointermove cycle?